src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -21,9 +21,29 @@
 
 namespace opossum {
 
-JitAwareLQPTranslator::JitAwareLQPTranslator() : LQPTranslator() {
+void resolve_aggregate_column_name_rec(const std::shared_ptr<opossum::LQPExpression>& expression,
+                                       std::stringstream& stream, const bool brackets = true) {
+  if (expression->type() == ExpressionType::Column) {
+    const auto& column_reference = expression->column_reference();
+    stream << column_reference.original_node()->output_column_names()[column_reference.original_column_id()];
+  } else {
+    if (brackets) stream << "(";
+    DebugAssert(expression->is_arithmetic_operator(), "Unsupported expression type");
+    resolve_aggregate_column_name_rec(expression->left_child(), stream);
+    stream << " " << expression_type_to_operator_string.at(expression->type()) << " ";
+    resolve_aggregate_column_name_rec(expression->right_child(), stream);
+    if (brackets) stream << ")";
+  }
 }
 
+std::string resolve_aggregate_column_name(const std::shared_ptr<opossum::LQPExpression>& expression) {
```

Comment:
I don't like the name "resolve" here. You don't look anything up, but you generate a new name. Maybe `column_name_for_aggregate`?

Reply:
Is renamed.